mods/bucket/init.lua

```diff
@@ -166,18 +171,21 @@ minetest.register_craftitem("bucket:bucket_empty", {
 			local source_neighbor = false
 			if liquiddef.force_renew then
 				source_neighbor =
-					minetest.find_node_near(pointed_thing.under, 1, liquiddef.source)
+					minetest.find_node_near(pos, 1, liquiddef.source)
 			end
-			if not (source_neighbor and liquiddef.force_renew) then
-				minetest.add_node(pointed_thing.under, {name = "air"})
+			if source_neighbor and liquiddef.force_renew then
+				log_action(pos, pname, "picked up " .. liquiddef.source .. " (force renewed)")
```

Comment:
I don't think logging the force-renewed or not here is that important. You can always check this by looking into the node def.

Reply:
I see, you're right.